pandas/core/groupby/generic.py

```diff
@@ -1848,6 +1848,7 @@ def nunique(self, dropna: bool = True):
             ],
             axis=1,
         )
+        assert isinstance(results, DataFrame)  # for mypy
```

Comment:
if you use cast here, then once we adopt Literal and concat can be overloaded on axis value, then with `warn_redundant_casts` we can remove

Reply:
neat, will do